### ISSUE TITLE
[analyzer][Extension types] No error if the representation type is a bottom type

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#53823]
According to the extension types feature [specification](https://github.com/dart-lang/language/blob/main/accepted/future-releases/extension-types/feature-specification.md)

> A compile-time error occurs if the representation type of an extension type declaration is a bottom type.

But there is no such error in analyzer
```dart
extension type ET1(Never _) {} // No error

extension type ET2<T extends Never>(T _) {} // No error
```
Related CFE issue #53824 

Tested on `Dart SDK version: 3.3.0-edge.849459109a364f7f80ca82647ad088a4a6157ebc (be) (Sat Oct 21 02:18:48 2023 +0000) on "macos_x64"`

**Comment by bwilkerson** [comment]
@scheglov

**Comment by scheglov** [comment]
I don't understand why you don't see errors here.
The change landed as 13791e4fee0833fb67018df155ca247bc2b6ae60.

I see it when try IntelliJ, and from cli.
```console
scheglov@scheglov-macbookpro4:~/Source/Dart/sdk.git/sdk (master)$ /Users/scheglov/Source/Dart/sdk.git/sdk/xcodebuild/ReleaseARM64/dart-sdk/bin/dart analyze /Users/scheglov/dart/test/bin/test.dart
Analyzing test.dart...                 0.4s

  error • test.dart:1:20 • The representation type can't be a bottom type. Try specifying a different type. • extension_type_representation_type_bottom
  error • test.dart:3:37 • The representation type can't be a bottom type. Try specifying a different type. • extension_type_representation_type_bottom

2 issues found.
scheglov@scheglov-macbookpro4:~/Source/Dart/sdk.git/sdk (master)$ /Users/scheglov/Source/Dart/sdk.git/sdk/xcodebuild/ReleaseARM64/dart-sdk/bin/dart --version
Dart SDK version: 3.3.0-edge.d37620ed0301133324e153349ec1eb7c26a559c0 (be) (Mon Oct 23 16:45:47 2023 +0000) on "macos_arm64"
```

**Comment by sgrekhov** [comment]
Retested on the freshly built SDK. No this issue in analyzer now. Probably I tested it on SDK without this change. Sorry for this. Closing as fixed
